python/ctsm/subset_data.py

```diff
@@ -0,0 +1,614 @@
+        single_point.create_datmdomain_at_point(file_dict["datm_tuple"])
+
+        # subset the DATM data
+        nl_datm = os.path.join(args.user_mods_dir, "user_nl_datm_streams")
+        single_point.create_datm_at_point(file_dict['datm_tuple'], args.datm_syr, args.datm_eyr,
+                                          nl_datm)
+
+    # -- Write shell commands
+    if single_point.create_user_mods:
+        single_point.write_shell_commands(os.path.join(args.user_mods_dir, "shell_commands"))
+
+    logger.info("Successfully ran script for single point.")
+
+
+def subset_region(args, file_dict: dict):
+    """
+    Subsets surface, domain, land use, and/or DATM files for a region
+    """
+
+    logger.info("----------------------------------------------------------------------------")
+    logger.info("This script extracts a region from the global CTSM datasets.")
+
+    # --  Create Region Object
+    region = RegionalCase(
+        args.lat1,
+        args.lat2,
+        args.lon1,
+        args.lon2,
+        args.reg_name,
+        args.create_domain,
+        args.create_surfdata,
+        args.create_landuse,
+        args.create_datm,
+        args.create_user_mods,
+        args.out_dir,
+    )
+
+    region.create_tag()
+    logger.debug(region)
+
+    # --  Create CTSM domain file
+    if region.create_domain:
+        region.create_domain_at_reg(file_dict["main_dir"], file_dict["fdomain_in"])
+
+    # --  Create CTSM surface data file
+    if region.create_surfdata:
+        region.create_surfdata_at_reg(file_dict["fsurf_dir"], file_dict["fsurf_in"],
+                                      args.user_mods_dir)
+
+    # --  Create CTSM transient landuse data file
+    if region.create_landuse:
+        region.create_landuse_at_reg(file_dict["fluse_dir"], file_dict["fluse_in"],
+                                     args.user_mods_dir)
+
+    logger.info("Successfully ran script for a regional case.")
+    sys.exit()
+
+
+def main():
+    """
+    Calls functions that subset surface, landuse, domain, and/or DATM files for a region or a
+    single point.
+    """
+
+    # add logging flags from ctsm_logging
+    setup_logging_pre_config()
+    parser = get_parser()
+    add_logging_args(parser)
+    args = parser.parse_args()
+    process_logging_args(args)
+
+    # parse defaults file
+    cesmroot = path_to_ctsm_root()
+    defaults = configparser.ConfigParser()
+    defaults.read(os.path.join(cesmroot, "tools/site_and_regional", DEFAULTS_FILE))
+
+    # --------------------------------- #
+
+    myname = getuser()
+    pwd = os.getcwd()
+    logger.info("User = %s", myname)
+    logger.info("Current directory = %s", pwd)
+
+    # --------------------------------- #
+
+    # print help and exit when no option is chosen
+    if args.run_type != "point" and args.run_type != "region":
+        print("Must supply a positional argument: 'point' or 'region'.")
+        print("See ./subset_data --help for more help.")
+        sys.exit()
```

Comment:
- [ ] This should use utils.abort(message) so that it aborts consistently with other code that signals an error condition.

Reply:
Answered in the [next](https://github.com/ESCOMP/CTSM/pull/1461#discussion_r780649519)

Reply:
This is changed to raise argparse parser.error as following:
``` python
        err_msg = textwrap.dedent('''\
                \n ------------------------------------
                \n Must supply one of:
                \n --create-surface \n --create-landuse \n --create-datm \n --create-domain \n
                \n See ./subset_data --help for more help.
                '''
                )   
        raise parser.error(err_msg)
```

Reply:
Perfect